### ISSUE TITLE
composer install results in NoSslException

### ISSUE THREAD

**Issue by LarsImNetz** [composer/composer#7596]
Hi,

I update composer from 1.6.5 to composer 1.7.2 and the result is that composer doesn't work any longer, see diagnose output.
With version 1.6.5 anything is ok.
I use a bash shell (Windows 10, git-bash 2.14.1)

Output of `composer diagnose`:

```
  [Composer\Exception\NoSslException]
  The openssl extension is required for SSL/TLS protection but is not available. If you can not enable the openss
  l extension, you can disable this error, at your own risk, by setting the 'disable-tls' option to true.

diagnose
```

openssl is installed and within composer 1.6.5 it works as expected.

```
php -m | grep openssl
openssl
```
I found out that we use the environment variable $PHP_INI_SCAN_DIR which shows to an alternative php.ini. In the composer shell script I see the line with php "${dir}/composer.phar" "$@" and when I add after php the parameter '-c $PHP_INI_SCAN_DIR' anything works again as expected.
I don't know why php in your script do not support such variable any longer. May be a bug.

Kind regards
Lars

**Comment by alcohol** [comment]
@johnstevenson could this be due to the xdebug handler?

**Comment by johnstevenson** [comment]
@LarsImNetz Have you got xdebug enabled? What version of PHP are you using? With Composer 1.7.2 you can troubleshoot xdebug-handler:

```
export XDEBUG_HANDLER_DEBUG=1
# outputs status messages

export XDEBUG_HANDLER_DEBUG=2
# as above, but saves the temporary ini that is used for the restarted process
```

**Comment by LarsImNetz** [comment]
Thank you @johnstevenson for the fast answer.

You are right, xdebug is enabled in php.ini.
If I disable xdebug, composer 1.7.2 works like charm. But we need composer xdebug. Is there a way to activate it the right way? Or I'am wrong I do not need xdebug any longer?

```
$ php -v
PHP 7.1.10 (cli) (built: Sep 26 2017 20:07:27) ( ZTS MSVC14 (Visual C++ 2015) x86 )
...
```

**Comment by alcohol** [comment]
Why do you need xdebug? Composer actively attempts to disable xdebug while running, because it decreases the performance of Composer significantly. We do this through our [xdebug-handler](/composer/xdebug-handler).

**Comment by LarsImNetz** [comment]
In real, I don't know why we need xdebug at composer time.

**Comment by LarsImNetz** [comment]
After I set the environment variable COMPOSER_ALLOW_XDEBUG=1. composer works as expected.

I think it's no longer an issue, so we can close it.

**Comment by johnstevenson** [comment]
@alcohol This is actually a bug in xdebug-handler (fix on its way). When the library changed its method of suppressing ini-scanning in the restart (to use simpler command-line options), I overlooked the fact that `php_ini_scanned_files` does not work if PHP is built without a --with-config-file-scan-dir value (Windows mainly).

To compound this idiocy, I was the one who reported the bug: [#73124](https://bugs.php.net/73124) (and even fixed it, after no one took any notice!). Fortunately, `PHP_INI_SCAN_DIR` is not generally a Windows thing.

@LarsImNetz There are a couple of other ways to handle this (until Composer gets updated). You could upgrade PHP (its bug was fixed in 7.1.13 and 7.2.1). Or if `PHP_INI_SCAN_DIR` points to a _single_ ini file, use `PHPRC` instead, which is more appropriate.

**Comment by alcohol** [comment]
I'll open this then. Once you fix it in the handler, submit a PR that references this issue :-)
